src/main/java/com/hivemq/mqtt/message/pool/FreePacketIdRanges.java

```diff
@@ -106,6 +106,12 @@ public void takeIfAvailable(final int id) {
                     }
                 }
 
+                // Updating the root range to preserve the invariant of root range being non-empty if there are ranges
+                // with free IDs available. We do so by ignoring intervals that consist of a single id.
+                while ((rootRange.start == rootRange.end) && (rootRange.next != null)) {
```

Comment:
"We do so by ignoring intervals that consist of a **single** id"
But if start == end, the interval is empty anyway, right?

Reply:
My bad, should be empty intervals in the comment